packages/frontend/src/model/theory_selector.tsx

```diff
@@ -48,22 +47,69 @@ export function TheorySelector(props: TheorySelectorProps) {
     const groupedTheories = createMemo(() =>
         Array.from(props.theories.groupedMetadata().entries()),
     );
-
     return (
         <div class="theory-selector">
             <For each={groupedTheories()}>
                 {([group, theories]) => (
                     <div class="group">
                         <div class="group-name">{group}</div>
-                        <For each={theories}>
+                        <For
+                            each={theories.filter((t) => {
```

Comment:
As a matter of style, I think it's more readable to define long helper functions like this before the JSX rather than inline, which due to the depth tends to force the formatter to chop up your code in unfortunate ways.